src/kubeseal_auto/kubeseal.py

```
@@ -600,27 +627,26 @@ def reencrypt(self, src: str) -> None:
 
                 # Create backup of original file
-                shutil.copy2(secret, backup_file)
+                shutil.copy2(secret, backup_path)
 
                 cmd = self._build_kubeseal_cmd(extra_args=["--re-encrypt"])
                 ic(cmd)
 
                 try:
-                    with open(str(secret)) as stdin_f, open(output_tmp, "w") as stdout_f:
+                    with secret.open() as stdin_f, output_tmp.open("w") as stdout_f:
                         subprocess.run(cmd, stdin=stdin_f, stdout=stdout_f, check=True)
```

Ruff (0.14.7)

`subprocess` call: check for execution of untrusted input

(S603)